docs/src/architecture/notes.md

```diff
@@ -1,82 +1,8 @@
 # Notes
-Miden aims to achieve parallel transaction execution and privacy. The UTXO-model combined with client-side proofs provide those features. In Miden, an note is a way of transferring assets between accounts. Notes can be consumed and produced asynchronously and privately.
+Miden aims to achieve parallel transaction execution and privacy. The UTXO-model combined with client-side proofs provide those features. That means, in Miden exist notes as a way of transferring assets between and to interact with accounts. Notes can be consumed and produced asynchronously and privately.
```

Comment:
It feels like most of this paragraph belongs in a higher-level section (e.g., Architecture chapter).

Reply:
Yes, and no. Last time, we wanted to describe on top of every sub-chapter the goals that we want to achieve with this particular design. 

So for notes, to me, it makes sense to state at the beginning of the Notes chapter what we want to achieve with Notes. Does that make sense to you?

![Screenshot 2023-12-05 at 12 24 04](https://github.com/0xPolygonMiden/miden-base/assets/35031754/075faa75-ca5d-41da-a8c1-65d360590322)

We have the same reasoning at the beginning of the Accounts chapter, see [here](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html#accounts)